fuel/datasets/hdf5.py

```diff
@@ -242,8 +271,16 @@ def load_axis_labels(self):
         handle = self._file_handle
         axis_labels = {}
         for source_name in handle:
-            axis_labels[source_name] = tuple(
-                dim.label for dim in handle[source_name].dims)
+            vlen_source = (len(handle[source_name].dims) == 1
```

Comment:
Sorry for spamming: seems like `source_name in self.sources_to_reshape` would be equivalent to `is_variable_length(source_name)` I suggest above.